SimExLite/SourceCalculators/GaussianSourceCalculator.py

```diff
@@ -0,0 +1,207 @@
+            parameters=parameters,
+        )
+        self.__WPG_path = None
+        self.WPG_path = WPG_path
+
+    @property
+    def WPG_path(self) -> str:
+        return self.__WPG_path
+
+    @WPG_path.setter
+    def WPG_path(self, value: str):
+        self.set_WPG_path(value)
+
+    def set_WPG_path(self, value: str):
+        if isinstance(value, str):
+            self.__WPG_path = value
+            if not Path(self.__WPG_path).is_dir():
+                raise ValueError("The input for WPG_path is not a valid directory.")
+            sys.path.insert(0, self.WPG_path)
+        else:
+            raise TypeError(
+                f"WPG_path is supposed to be in type str, instead of {type(value)}"
+            )
+
+    def init_parameters(self):
+        parameters = CalculatorParameters()
+
+        photon_energy = parameters.new_parameter(
+            "photon_energy", comment="The mean photon energy", unit="eV"
+        )
+        photon_energy.value = 8e3
+
+        relative_bandwidth = parameters.new_parameter(
+            "photon_energy_relative_bandwidth", comment="The relative energy bandwidth"
+        )
+        relative_bandwidth.add_interval(0, None, True)
+        relative_bandwidth.value = 1e-3
+
+        diameter = parameters.new_parameter(
+            "beam_diameter_fwhm", comment="Beam diameter", unit="m"
+        )
+        diameter.value = 1e-4
+
+        pulse_energy = parameters.new_parameter(
+            "pulse_energy", comment="Total energy of the pulse", unit="joule"
+        )
+        pulse_energy.value = 2e-3
+
+        divergence = parameters.new_parameter(
+            "divergence", comment="Beam divergence angle", unit="radian"
+        )
+        divergence.add_interval(0, 2 * np.pi, True)
+        diver_value = get_divergence_from_beam_diameter(
+            photon_energy.value, diameter.value
+        )
+        divergence.value = diver_value
+
+        spectrum_type = parameters.new_parameter(
+            "photon_energy_spectrum_type",
+            comment='Type of energy spectrum ("SASE" | "tophat" | "twocolour", default "SASE")',
+        )
+        spectrum_type.value = "SASE"
+
+        number_of_transverse_grid_points = parameters.new_parameter(
+            "number_of_transverse_grid_points",
+            comment="The number of grid points in both horizontal (x) and vertical (y) dimension transverse to the beam direction",
+        )
+        number_of_transverse_grid_points.add_interval(0, None, True)
+        number_of_transverse_grid_points.value = 400
+
+        num_time_slices = parameters.new_parameter(
+            "number_of_time_slices", comment="The number of time slices"
+        )
+        num_time_slices.add_interval(0, None, True)
+        num_time_slices.value = 12
+
+        param_z = parameters.new_parameter(
+            "z", comment="The position of the pulse in the beam direction", unit="m"
+        )
+        param_z.value = 100
```

Comment:
The z position should be a parameter of the calculator, ie. the user can set it.

Reply:
no thats totally fine. but here the z value is set by the calculator itself. i'm saying z should be coming from the code that constructs/configures the calculator

Reply:
Do you mean something like:
`calculator = GaussianSourceCalculator("gaussian_source", parameters=parameters, z=100)`?
I would think that "z" one of the `parameters`. Wouldn't that be a bit confusing?